### DISCUSSION TITLE
如何通过ReplyChain获取回复的消息?

### DISCUSSION THREAD

**Discussion by dawn-lc** [KonataDev/Konata.Core#162] [General]
我尝试建立一个缓存 MessageStruct 表 ,但是 ReplyChain 的 UUID 和 MessageStruct  的 UUID 并不一致

**Comment by dawn-lc** [comment]
用 Sequence 来找就行了,但是 ReplyChain 的 Sequence 不可访问, 是否应该更改访问修饰符?

**Comment by lepollo** [comment]
不确定replyChain的tostring是否包含UUID，但确定包含信息内容和发送者
我直接提取了发送者QQ和文本内容，例子：
private void DoGroupMessage(Bot sender, GroupMessageEvent e)
{
	ReplyChain? rc = e.Chain.FindChain<ReplyChain>().FirstOrDefault(); //提取被回复的信息
          if (rc != null)
          {
                    string str = rc.ToString(); //这是一个json格式的信息
                    string msg = str.Substring(str.IndexOf("content")).Replace("content=", "").Replace("]}", ""); 	//图简单直接提取被回复信息的文本内容
                    string qq= str.Substring(str.IndexOf("qq"), str.IndexOf(",seq") - str.IndexOf("qq")).Replace("qq=", "").Trim();	//被回复信息的发送者
            }
}

**Reply by dawn-lc** [reply]
> 不确定replyChain的tostring是否包含UUID，但确定包含信息内容和发送者 我直接提取了发送者QQ和文本内容，例子： private void DoGroupMessage(Bot sender, GroupMessageEvent e) { ReplyChain? rc = e.Chain.FindChain().FirstOrDefault(); //提取被回复的信息 if (rc != null) { string str = rc.ToString(); //这是一个json格式的信息 string msg = str.Substring(str.IndexOf("content")).Replace("content=", "").Replace("]}", ""); //图简单直接提取被回复信息的文本内容 string qq= str.Substring(str.IndexOf("qq"), str.IndexOf(",seq") - str.IndexOf("qq")).Replace("qq=", "").Trim(); //被回复信息的发送者 } }

ReplyChain 消息内部并不包含回复所引用的消息

**Reply by dawn-lc** [reply]
需要建立MessageStruct缓存, 提取seq然后在缓存里找回复所引用的原消息

**Reply by dawn-lc** [reply]
content字段里至少不包含ImageChain

**Comment by dawn-lc** [comment]
https://github.com/dawn-lc/Nanako
我这的里面实现了对图片进行回复并ATbot进行搜图的功能

**Comment by takayama-lily** [comment]
暂时无法获取，除非你自己建立数据库存放消息并查询。

**Reply by TheSnowfield** [reply]
是的 core 不提供資料庫功能, 故無法實現這個需求,
需要自己對接收到的訊息儲存以在需要時調用。

**Reply by dawn-lc** [reply]
希望uuid可以统一一下, 方便作为key
